### PR TITLE
Add top level summary to QPY module docs

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds a missing top level description of what QPY is to the module docs for QPY. The existing documentation provides a good overview of how to use qpy and a very detailed guide to the format specification, but it does not actually say what QPY is up front. This commit fixes that gap and also adds an example for saving multiple circuits in a single file.

### Details and comments

### CODE REVIEW

File: qiskit/qpy/__init__.py [hunk 1]
```diff
@@ -17,6 +17,12 @@
 
 .. currentmodule:: qiskit.qpy
 
+QPY is a binary serialization format for :class:`~.QuantumCircuit` and
+:class:`~.ScheduleBlock` objects that is designed to be cross-platform,
+Python version agnostic, and backwards compatible moving forward. QPY should
+be used if you need a lossless mechanism to save or copy between systems a
```
Comment (nkanazawa1989) [inline qiskit/qpy/__init__.py:23]: For end-users, it would be better to describe how this "lossless" use case differs from qasm3 exporter/importer?

**Review by nkanazawa1989 — APPROVED** [review]
LGTM, thanks :)

File: qiskit/qpy/__init__.py [hunk 2]
```diff
@@ -42,6 +57,17 @@
     with open('bell.qpy', 'rb') as fd:
         new_qc = qpy.load(fd)[0]
 
+the :func:`qiskit.circuit.qpy_serialization.dump` function also lets you
```
Comment (jakelishman) [inline qiskit/qpy/__init__.py:60]: This path is outdated - should be `qiskit.qpy.dump`. I think there's several cases throughout the documentation that need updating as well.  Do you want to do that in this PR, or in a separate one?

(This sentence also needs to start with a capital letter.)

Comment (mtreinish) [reply qiskit/qpy/__init__.py:60]: I'll fix it now, it's easy to do and it'll be good to get this in the same backport PR.

Comment (mtreinish) [reply qiskit/qpy/__init__.py]: Fixed in: https://github.com/Qiskit/qiskit-terra/pull/9212/commits/c07be51deb127505a5a6a9641bae9e45139aaba2

**Review by jakelishman — APPROVED** [review]

**Pull request merged**